streams/src/main/java/org/apache/kafka/streams/state/internals/ReadonlyPartiallyDeserializedSegmentValue.java

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.state.internals;
+
+import org.apache.kafka.streams.query.ResultOrder;
+
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.Map;
+
+
+final class ReadonlyPartiallyDeserializedSegmentValue {
+
+    private static final int TIMESTAMP_SIZE = 8;
+    private static final int VALUE_SIZE = 4;
+    private byte[] segmentValue;
+    private long nextTimestamp;
+    private long minTimestamp;
+
+    private int deserIndex = -1; // index up through which this segment has been deserialized (inclusive)
+
+    private Map<Integer, Integer> cumulativeValueSizes;
+
+    private int valuesStartingIndex = -1; // the index of the first value in the segment (but the last one in the list)
+    private Map<Integer, TimestampAndValueSize> unpackedTimestampAndValueSizes = new HashMap<>();
+    private int recordNumber = -1; // number of segment records
+
+
+    ReadonlyPartiallyDeserializedSegmentValue(final byte[] segmentValue) {
+        this.segmentValue = segmentValue;
+        this.nextTimestamp =
+                RocksDBVersionedStoreSegmentValueFormatter.getNextTimestamp(segmentValue);
+        this.minTimestamp =
+                RocksDBVersionedStoreSegmentValueFormatter.getMinTimestamp(segmentValue);
+        resetDeserHelpers();
+    }
+
+
+    public long minTimestamp() {
+        return minTimestamp;
+    }
+
+    public long nextTimestamp() {
+        return nextTimestamp;
+    }
+
+    public byte[] serialize() {
```

Comment:
> Do we need this method? Cannot find where it's used (except test code)

The test code is also important. We may future usage as well. What 's the problem of having it?

Reply:
Yes tests are important, but the test you added is not testing this method, but only adds a redundent test for `getTimestamp()`? Seem we don't need this newly added test a second time?

> We may have future usage as well.
We would add it when we need it :) 

> What's the problem with having it?
It's about "cleaness" -- we also don't add a method that computes the Fibonacci sequence, because we might need it in the future :)